unity-environment/Assets/ML-Agents/Scripts/Academy.cs

```diff
@@ -216,24 +239,75 @@ void InitializeEnvironment()
         // Retrieve Brain and initialize Academy
         List<Brain> brains = GetBrains(gameObject);
         InitializeAcademy();
-
-        // Check for existence of communicator
-        communicator = new ExternalCommunicator(this);
-        if (!communicator.CommunicatorHandShake())
+        MLAgents.Communicator.Communicator communicator= null;
+        try
+        {
+            communicator = new MLAgents.Communicator.RpcCommunicator(
+                new MLAgents.Communicator.CommunicatorParameters
+                {
+                Port = ReadArgs()
+                });
+        }
+        catch
         {
             communicator = null;
+            foreach (Brain b in brains)
```

Comment:
Are you creating several communicators here?
The logic in this try/catch is not very clear. A comment would help.